enterprise/frontend/src/embedding-sdk/store/auth/auth.ts

```diff
@@ -110,89 +91,58 @@ export const refreshTokenAsync = createAsyncThunk(
-        if ("message" in session && typeof session.message === "string") {
-          // For some errors, the BE gives us a message that explains it
-          throw new Error(session.message);
-        }
-        if (typeof session.status === "string") {
-          // other times it just returns an error code
-          throw new Error(
-            `Failed to refresh token, got status: ${session.status}`,
-          );
-        }
+    if (!session || typeof session !== "object") {
+      throw MetabaseError.INVALID_SESSION_OBJECT({
+        expected: "{jwt: string}",
+        actual: JSON.stringify(session),
+      });
+    }
+    if ("status" in session && session.status !== "ok") {
+      if ("message" in session && typeof session.message === "string") {
+        throw MetabaseError.INVALID_SESSION_OBJECT({
+          expected: "{jwt: string}",
+          actual: session.message,
+        });
       }
-      // Lastly if we don't have an error message or status, check if we actually got the session ID and expiration
-      if (!sessionSchema.isValidSync(session)) {
-        throw new Error(
-          `The ${source} must return an object with the shape {id:string, exp:number, iat:number, status:string}, got ${safeStringify(session)} instead`,
-        );
+      if (typeof session.status === "string") {
+        throw MetabaseError.INVALID_SESSION_OBJECT({
+          expected: "{jwt: string}",
+          actual: session.status,
+        });
       }
-      return session;
-    } catch (exception: unknown) {
-      // The host app may have a lot of logs (and the sdk logs a lot too), so we
-      // make a big red error message to make it visible as this is 90% a blocking error
-      console.error(...bigErrorHeader("Failed to get auth session"), exception);
-
-      throw exception;
     }
+    if (!sessionSchema.isValidSync(session)) {
+      throw MetabaseError.INVALID_SESSION_SCHEMA({
+        expected: "{id:string, exp:number, iat:number, status:string}",
+        actual: JSON.stringify(session),
+      });
+    }
+    return session;
   },
 );
 
-const safeStringify = (value: unknown) => {
-  try {
-    return JSON.stringify(value);
-  } catch (e) {
-    return value;
-  }
-};
-
 const getRefreshToken = async (
   url: MetabaseAuthConfig["metabaseInstanceUrl"],
   customFetchRequestToken:
     | MetabaseAuthConfig["fetchRequestToken"]
     | null = null,
 ) => {
-  // GET /auth/sso with headers
-  const urlResponse = await fetch(`${url}/auth/sso`, getSdkRequestHeaders());
-  const urlResponseJson = await urlResponse.json();
-
-  // For the SDK, both SAML and JWT endpoints return {url: [...], method: "saml" | "jwt"}
-  // when the headers are passed
-  const { method, url: responseUrl, hash } = urlResponseJson;
-
+  let urlResponseJson;
+  try {
+    urlResponseJson = await connectToInstanceAuthSso(url);
+  } catch (e) {
+    console.error("connectToInstanceAuthSso threw:", e);
+    throw e;
+  }
```

Comment:
not sure if the `console.error("connectToInstanceAuthSso threw")` is meant to be for debugging as we are re-throwing the error anyway, and it's accidentally left in

```suggestion
  const urlResponseJson = await connectToInstanceAuthSso(url);
```

Reply:
that's leftovers from debugging